p2p/swarm.go

```diff
@@ -95,19 +95,24 @@ func newSwarm(config config.Config, newNode bool) (*swarm, error) {
 
 	s.lNode.Debug("Created swarm for local node %s, %s", l.Address(), l.Pretty())
 
+
+
+	return s, nil
+}
+
+func (s *swarm) Start() error{
 	go s.listenToNetworkMessages()
 
-	if config.SwarmConfig.Bootstrap {
+	if s.config.SwarmConfig.Bootstrap {
 		err := s.dht.Bootstrap()
```

Comment:
requirement was to add channels before bootsrapping, not to make bootsrap run in background, which i think is wrong to do, this is the foundation of the system if bootsrap doesn't succeed all other actions are pointless

Reply:
Please read the issue description again. 
To the point, the node can operate it degraded mode without completing the bootstrap and it can halt in case of bootstrap failure, it has nothing to do with the context in which the bootstrap is running in.
In current state the bootstrap process blocks the nodes start sequence, for example during bootstrap the nodes doesn't listen to abort signals...